rejson.go

```diff
@@ -17,8 +17,14 @@ func NewReJSONHandler() *Handler {
 type ReJSON interface {
 	JSONSet(key, path string, obj interface{}, opts ...rjs.SetOption) (res interface{}, err error)
 
+	JSONSetWithIndex(key, path string, obj interface{}, index string) (res interface{}, err error)
+
 	JSONGet(key, path string, opts ...rjs.GetOption) (res interface{}, err error)
 
+	JSONQGet(key string, params ...string) (res interface{}, err error)
+
+	JSONIndexAdd(key, field, path string) (res interface{}, err error)
+
```

Comment:
Instead of adding these methods in `ReJSON` interface create a separate interface `ReJSON2` and embed `ReJSON` interfaces in it along with the new methods. And create a separate handler for the ReJSON2, so that the new module won't affect the original one.

Something like this:
```suggestion
type ReJSON2 interface {
	ReJSON
	JSONSetWithIndex(key, path string, obj interface{}, index string) (res interface{}, err error)
	JSONQGet(key string, params ...string) (res interface{}, err error)
	JSONIndexAdd(key, field, path string) (res interface{}, err error)
}
```